### ISSUE TITLE
History Import Timestamps (import date, vs watched date)

### ISSUE THREAD

**Issue by metal450** [sniklaus/youtube-watchmarker#13]
When first installing this extension & importing previous watch history, it seems to set the watch timestamps to the current time (aka the import time), rather than the actual watch time.

**Comment by sniklaus** [comment]
Thank you for bringing this up! When synchronizing with the browser history, it will use the time at which you have last visited the page of the video according to your browser. When synchronizing with the Youtube history, it will use the current time for newly discovered videos. There unfortunately no easy solution to using the actual timestamp when syncing with the Youtube history since this is information that they do not expose. I am hence closing this issue for now but will post an update in case I ever find a better solution.

**Comment by metal450** [comment]
>>There unfortunately no easy solution to using the actual timestamp when syncing with the Youtube history since this is information that they do not expose.

Isn't that information visible (and thus parsable) when vising the 'history' page on YouTube.com?  I do see it listed above each group of videos.

**Comment by sniklaus** [comment]
But that categorization is only on a per-day basis, so it is not sufficient to pinpoint the actual time when a video has been watched.

**Comment by metal450** [comment]
True. But still *significantly* more informative than having no idea when it was viewed at all ;)

**Comment by sniklaus** [comment]
Are the categories consistent though? Or does it switch to "last month" or "last year" once you go back sufficiently far? It is also a little tedious to implement this so I am afraid that I am holding off from implementing it based on incomplete information.

**Comment by metal450** [comment]
The naming convention is well-defined.  There are exactly 4 formats, so 4 pattern matches needed:
Today|Yesterday (this style is for 2 days)
DayOfTheWeek (this style is for the next 5 days)
Mar 25 (this style is through the beginning of the current year)
Dec 25, 2019 (ongoing)

The other reason this would be much better is because, as I understand it, it would cause multiple PCs to have consistent histories.  As it is currently implemented, importing from 2 different PCs would cause them to have different histories (because each would show the watch date whenever that PC happened to sync last).  If it was based on the parsed info actually shown, they would always be the same, regardless of when that PC last imported.

**Comment by sniklaus** [comment]
That is good to know, but I am afraid that I am not willing to invest my spare time into adding this feature at the moment, my apologies. I would be happy to review a pull request that adds this feature though.

**Comment by metal450** [comment]
Understood. I've never seen even a line of browser addon code, so I'm afraid it would require some studying on my part too, which I'm unfortunately not in place to do at this particular moment.  Still, I'd propose not closing this issue, as it's not actually solved, & perhaps one of us (or someone else looking at these issues in the future) might come along & agree with the idea ;)

**Comment by sniklaus** [comment]
Fair enough, reopening. Anyone who wants to give it a shot, the part that would need to be updated is quoted below.

https://github.com/sniklaus/youtube-watchmarker/blob/1c79ffda2d1960eb20daa17c3694972bb99dbae3/background.js#L624-L685

**Comment by yutotakano** [comment]
Hi folks, I've built a web tool based that converts a JSON Google Takeout of your YouTube watch history into the database format expected by youtube-watchmarker.

It builds on the [tool](https://github.com/janpaul123/youtube-takeout-to-watchmarker) linked in the README which converts HTML exports, as well as an attempted [port](https://github.com/Jeefongithub/youtube-takeout-json-to-watchmarker) two years ago to JSON exports. Neither supported maintaining timestamps, and the technical prerequisites to the tool were quite high, so I've made a modified version.

Mine maintains the watch timestamps provided by YouTube, so I'm quite happy with the result. I'm sharing it here in hopes that someone else might stumble upon the thread in the future!

Repo: https://github.com/yutotakano/youtube-takeout-json-to-watchmarker
Tool: https://yutotakano.github.io/youtube-takeout-json-to-watchmarker/
